### PR TITLE
Add -feature and -language:reflectiveCalls to compilerArgs in conf/application.conf. Fixed compiler warnings.

### PR DESCRIPTION
**Pull request by deanwampler**
This pull request modified `compilerArgs` in `conf/application.conf`. It adds `-feature`, to explain when optional language features are used without explicitly enabling them, and it adds `-language:reflectiveCalls` to explicitly enable this optional feature, which used by Spark Notebook itself.

Without both compiler flags, almost every cell prints a warning from the compiler about an optional feature being used, but with no useful information except to add `-feature` to get more details. The actual issue is the use reflection by Spark Notebook itself.

### CODE REVIEW

**Comment by deanwampler** [comment]
I pushed some other changes too soon to the same branch. These changes fix some compiler warnings, especially catch clauses that catch fatal exceptions, but should use `NonFatal` instead.

File: conf/application.conf [hunk 1]
```diff
@@ -139,8 +139,13 @@ manager {
...
     # REPL compiler options: Use the deprecation warning by default for more
-    # useful feedback about obsolute functions, etc.
-    compilerArgs=[-deprecation]
+    # useful feedback about obsolete functions, etc.
+    # REPL compiler options: Use the -deprecation warning by default for more
+    # useful feedback about obsolete functions, etc. Use the -feature warning
+    # for more explicit warnings about "optional" language features that
+    # should be enabled explicitly. One of those that's used by Spark Notebook
+    # itself is "reflective calls".
+    compilerArgs=["-deprecation", "-feature", "-language:reflectiveCalls"]
```
Comment (andypetrella) [inline conf/application.conf]: :cool:

**Comment by andypetrella** [comment]
LGTM!
thanks a lot @deanwampler

File: modules/spark/src/main/scala_2.10/spark-pre1.5/notebook/kernel/Repl.scala [hunk 2]
```diff
@@ -252,7 +252,7 @@ class Repl(val compilerOpts: List[String], val jars:List[String]=Nil) extends Re
                 }
                 iws(o)
               } catch {
-                case e =>
+                case NonFatal(e) =>
```
Comment (vidma) [inline modules/spark/src/main/scala_2.10/spark-pre1.5/notebook/kernel/Repl.scala]: nice!

**Pull request merged**